### ISSUE TITLE
Typescript errors for example ml_beginners

### ISSUE THREAD

**Issue by QuantumInformation** [PAIR-code/deeplearnjs#754]
I'm trying one of your examples: 
https://github.com/PAIR-code/deeplearnjs/blob/master/docs/tutorials/ml_beginners.md

snadbox here: 

https://github.com/QuantumInformation/deeplearnjs-playground/blob/master/src/index.ts#L91

However I am getting some typescript errors like:
```
ERROR in [at-loader] ./src/index.ts:91:26
    TS2345: Argument of type '() => Tensor<Rank>' is not assignable to parameter of type '() => Scalar'.
  Type 'Tensor<Rank>' is not assignable to type 'Scalar'.
    Types of property 'rankType' are incompatible.
      Type 'Rank' is not assignable to type 'Rank.R0'.
```

is the example out of date with the 0.5 api?

**Comment by dsmilkov** [comment]
That example is written in es6, and lacks any typescript annotations. If you want to port it to TypeScript, you'll have to annotate the types.

The error in question can be resolved by marking the result of the function `f` passed to `optimizer.minimize` as dl.`Scalar`. That is:

```ts
optimizer.minimize(() => {
  // Feed the examples into the model
  const pred = predict(xs[i]);
  const predLoss = loss(pred, ys[i]);
  ///////// Mark as scalar. ///////
  return predLoss as dl.Scalar; 
});
```

You might want into other issues as well, mostly asking you to type tensors as specific rank. Let me know how it goes.

**Comment by ManrajGrover** [comment]
This has been fixed in #761. I believe we can close this.

**Comment by QuantumInformation** [comment]
Yes thx

Code here https://github.com/QuantumInformation/deeplearnjs-playground/tree/master/quadraticEq
